### PR TITLE
🐛 Fixes modifier override order for defaultStepOptions

### LINKED ISSUE #816: defaultStepOptions.popperOptions only seems to apply to first step
Given the following input:
```js
 const tour = new Shepherd.Tour({
    tourName: "Demo",
    // ...
    defaultStepOptions: {
      scrollTo: { behavior: "smooth", block: "center" },
      popperOptions: {
        modifiers: [{ name: "offset", options: { offset: [0, 32] } }]
      },
    }
  });

 // ...
 tour.addSteps([
   // ...
 ]);
```

`defaultStepOptions.popperOptions` (i.e. the popper offset) only seems to apply to the first step in the tour. The other default option, `scrollTo`, applies to each step just fine.

Any insights?

### PR DESCRIPTION
**Pull request by chuckcarpenter**
closes #816

### CODE REVIEW

File: src/js/step.js [hunk 1]
```diff
@@ -289,6 +289,7 @@ export class Step extends Evented {
    * @private
    */
   _setOptions(options = {}) {
+
```
Comment (rwwagner90) [inline src/js/step.js]:
```suggestion
```

File: src/js/utils/general.js [hunk 2]
```diff
@@ -118,15 +118,10 @@ export function getPopperOptions(attachToOptions, step) {
         (mod) => !names.includes(mod.name)
       );
 
-      popperOptions.modifiers = [
-        ...step.options.popperOptions.modifiers,
-        ...filteredModifiers
-      ];
-
-      delete step.options.popperOptions.modifiers;
+      popperOptions.modifiers = Array.from(new Set([...popperOptions.modifiers, ...filteredModifiers, ...step.options.popperOptions.modifiers]));
```
Comment (rwwagner90) [inline src/js/utils/general.js]: What's the browser support on `Array.from`?

File: landing/js/welcome.js [hunk 3]
```diff
@@ -18,6 +18,9 @@
         scrollTo: {
           behavior: 'smooth',
           block: 'center'
+        },
+        popperOptions: {
+          modifiers: [{ name: "offset", options: { offset: [0, 32] } }]
```
Comment (rwwagner90) [inline landing/js/welcome.js]: Do we want this offset on the landing or was it leftover from where you were playing with it?

Comment (chuckcarpenter) [reply landing/js/welcome.js]: I think it's nice to have these examples in while developing, but I can remove if you prefer.

Comment (rwwagner90) [reply landing/js/welcome.js]: Does this not change the style of the landing page though? I think we would want to keep the landing page styled as it was, right?

File: src/js/utils/general.js (hunk 2 above)
Comment (chuckcarpenter) [reply src/js/utils/general.js]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from#Browser_compatibility

**Pull request merged**